extensions/swiss-ov/src/search-connection.tsx

```diff
@@ -0,0 +1,129 @@
+import { Form, ActionPanel, Action } from "@raycast/api";
+import { useNavigation } from "@raycast/api";
+import { Color, Icon, List } from "@raycast/api";
+import { useFetch } from "@raycast/utils";
+import moment = require("moment");
+
+type Values = {
+  from: string;
+  to: string;
+  isArrival: boolean;
+  date: Date;
+};
+
+type OVConnection = {
+  transfers: string;
+  duration: string;
+  from: {
+    departureTimestamp: string;
+    station: { name: string };
+    departure: string;
+    platform?: string;
+  };
+  to: {
+    arrival: string;
+    arrivalTimestamp: string;
+    station: { name: string };
+  };
+  products: string[];
+  sections: [
+    {
+      departure: { station: { name: string }; departure: string; platform?: string };
+      arrival: { station: { name: string }; arrival: string; platform?: string };
+    },
+  ];
+};
+
+function getConnectioMarkdown(connection: OVConnection): string {
+  const dateFormat = "HH:mm,  DD.MM.YYYY";
+  return `# ${connection.from.station.name} to ${connection.to.station.name}
+Departure: ${moment(connection.from.departure).format(dateFormat)} 
+
+(in ${moment.duration(moment(connection.from.departure).diff(new Date())).humanize()}),
+
+Plattform: ${connection.from.platform || "unknown"}
+
+${connection.products.join(", ")}  
+
+${
+  connection.sections.length <= 1
+    ? ""
+    : connection.sections
+        .map(
+          (section) =>
+            `## from ${section.departure.station.name} to ${section.arrival.station.name}
+
+Departure: ${moment(section.departure.departure).format(dateFormat)}, Plattform ${section.departure.platform || "unknown"}
+
+Arrival: ${moment(section.arrival.arrival).format(dateFormat)}, Plattform ${section.arrival.platform || "unknown"}
+
+`,
+        )
+        .join("\n")
+}
+`;
+}
+
+export default function Command() {
+  const { push } = useNavigation();
+  function handleSubmit(values: Values) {
+    push(<Connections {...values} />);
+  }
+
+  return (
+    <Form
+      actions={
+        <ActionPanel>
+          <Action.SubmitForm onSubmit={handleSubmit} />
+        </ActionPanel>
+      }
+    >
+      <Form.Description text={"Swiss Public Transport Connection"} />
+      <Form.TextField id="from" title="From" placeholder="From" storeValue />
+      <Form.TextField id="to" title="To" placeholder="To" storeValue />
```

Comment:
Is there a way to use a dropdown with values instead? Pure text might be hard?

Reply:
You don't need to enter the full name of the station. ZIP codes will work too. The API will try to find the best match.

I think people will use it to get home from the office (on the road, there are better mobile solutions) and therefore will use the same from-to entries. To help with that, the extension will load the last entries for "from" and "to."

But of course, I see your point. Maybe in the next release, I could add the possibility to save favorite stations.